plugins/search-react/src/wiring/createSearchResultListItemExtension.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { lazy, Suspense } from 'react';
+
+import { ListItemProps } from '@material-ui/core';
+
+import {
+  ExtensionBoundary,
+  createExtension,
+  createExtensionDataRef,
+} from '@backstage/frontend-plugin-api';
+import { Progress } from '@backstage/core-components';
+import { SearchDocument, SearchResult } from '@backstage/plugin-search-common';
+
+import { SearchResultListItemExtension } from '../extensions';
+
+/** @alpha */
+export type SearchResultItemExtensionComponent = <
+  P extends BaseSearchResultListItemProps,
+>(
+  props: P,
+) => JSX.Element | null;
+
+/** @alpha */
+export type SearchResultItemExtensionPredicate = (
+  result: SearchResult,
+) => boolean;
+
+/** @alpha */
+export const searchResultItemExtensionData = createExtensionDataRef<{
+  predicate?: SearchResultItemExtensionPredicate;
+  component: SearchResultItemExtensionComponent;
+}>('plugin.search.result.item.data');
+
+/** @alpha */
+export type SearchResultItemExtensionOptions = {
+  /**
+   * The extension id.
+   */
+  id: string;
+  /**
+   * The extension attachment point (e.g., search modal or page).
+   */
+  at: string;
+  /**
+   * The extension component.
+   */
+  component: () => Promise<SearchResultItemExtensionComponent>;
+  /**
+   * When an extension defines a predicate, it returns true if the result should be rendered by that extension.
+   * Defaults to a predicate that returns true, which means it renders all sorts of results.
+   */
+  predicate?: SearchResultItemExtensionPredicate;
+};
+
+/** @alpha */
+export type BaseSearchResultListItemProps = {
+  rank?: number;
```

Comment:
are noTrack and rank something that the integrator potentially want to configure or is the extension author the person that should decide?

Reply:
Good catch, it should be configurable by the integrator! I'll adjust it, thanks for pointing out 🙂

Reply:
Fixed here 🙌🏻 : https://github.com/backstage/backstage/pull/20137/commits/b5535ee53489301e2a45d5eeaefc1439d75a671d